packages/fxa-auth-server/scripts/recorded-future/check-and-reset.ts

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A script to check Recorded Future's Identity API for leaked credential, and
+ * force an account reset on accounts with valid leaked credentials.
+ *
+ * This script relies on the same set of enivronment variables as the FxA auth
+ * server.
+ */
+
+/**
+ * The general steps are:
+ *   1. Query Recorded Future's Identity credential search API endpoint to
+ *   fetch new leaked credential since the previous check.
+ *   2. For logins that are email addresses with an account in FxA, query
+ *   credential lookup API endpoint to get the leaked credentials.
+ *   3. For accounts with valid leaked plaintext passwords and no TOTP 2FA,
+ *   reset the account.
+ *   4. Send an email to the affected account holder to reset their password.
+ */
+
+import { promisify } from 'util';
+
+import { Command } from 'commander';
+import { StatsD } from 'hot-shots';
+import createClient from 'openapi-fetch';
+
+import { parseBooleanArg } from '../lib/args';
+import { emitStatsdMetrics } from '../lib/metrics';
+import appConfig from '../../config';
+
+import { paths } from './identity-schema';
+
+const config = appConfig.getProperties();
+const searchDomain = 'accounts.firefox.com';
+const statsd = new StatsD({ ...config.statsd });
+const client = createClient<paths>({
+  baseUrl: 'https://api.recordedfuture.com',
+  headers: { 'X-RFToken': config.recordedFuture.identityApiKey },
+});
+
+const checkAndReset = async () => {
+  const program = new Command();
+  program
+    .description(
+      `Query Recorded Future's Identity API for leaked account credentials.  Accounts
+with valid leaked credentials and without TOTP 2FA will be reset.  An email
+will be sent to the account holder.`
+    )
+    .option(
+      '--first-downloaded-date <date>',
```

Comment:
Is this a "search from" date that would correspond to the last time the script was run?

Reply:
Yeah.  New credentials since the last run.

Reply:
I squinted at this a bit - not sure if a different name might help. `last-checked-date` or `start-check-from-date` maybe?

But 🤷‍♀️ naming

Reply:
I see.  It's based on the first_downloaded_gte filter from the API endpoint.  I'll let this tumble in my head for a bit to see if I can think of something better.  But we can only convey so much context in the few words of parameter name.